src/mpi/coll/allreduce/allreduce.c

```diff
@@ -183,6 +183,47 @@ int MPIR_Allreduce_impl(const void *sendbuf, void *recvbuf, int count, MPI_Datat
                         MPI_Op op, MPIR_Comm * comm_ptr, MPIR_Errflag_t * errflag)
 {
     int mpi_errno = MPI_SUCCESS;
+    MPL_pointer_attr_t send_attr, recv_attr;
+    int sendbuf_in_device = 0, recvbuf_in_device = 0;
```

Comment:
style nitpicks: I'd prefer `bool` types and `if (buf_in_device)` comparisons, i.e. no `== 1`.

Reply:
Updated.